marine.c

```diff
@@ -579,6 +579,35 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
     return result;
 }
 
+
+WS_DLL_PUBLIC int compile_bpf(char *bpf, struct bpf_program *fcode, char *err_msg) {
+    pcap_t *pc;
+
+    pc = pcap_open_dead(DLT_EN10MB, MIN_PACKET_SIZE);
+    if (pc != NULL) {
```

Comment:
It's better to do:
```c
if (ERROR) {
   // hande_error
   return -1;
}
// main logic
```

than doing (like you did):
```c
if (NO_ERROR) {
    // main logic
   return 0;
}
// handle error
```
.

That way your function is more straight forward and has less indentations.